addons/delete-others/userscript.js

```diff
@@ -0,0 +1,64 @@
+export default async function ({ addon, console, msg }) {
+  const vm = await addon.tab.traps.vm;
+
+  const types = ["sound", "costume"];
+  let deletedItems = [];
+  let deleted;
+  let target;
+
+  function getRestoreFun(type) {
+    if (type === "costume") {
+      deletedItems.forEach((costume) => {
+        vm.addCostume(costume.md5, costume, target.id, "");
+      });
+    } else if (type === "sound") {
+      deletedItems.forEach((sound) => {
+        vm.addSound(sound, target.id);
+      });
+    }
+    deletedItems = [];
+  }
+
+  addon.tab.createEditorContextMenu(
+    (ctx) => {
+      if (window.confirm(`Are you sure you want to delete all other ${ctx.type}s?`)) {
+        const type = ctx.type === "costume" ? "Costume" : "Sound";
+
+        deletedItems = [];
+        target = vm.editingTarget;
+
+        const loop = ctx.type === "costume" ? target.getCostumes().length - 1 : target.getSounds().length - 1;
+
+        for (let i = loop; i > -1; i--) {
```

Comment:
Using a decrementing loop here seems a little weird. I see a `for...in` loop wasn't used instead though.

Reply:
Actually you could use a `for..of` loop here.

Reply:
Funny, @Joeclinton1 was adamant it was the preferred solution here, and after using it, I do agree. There is no array of assets to loop over as that just doesn't make sense, and due to the fact that deleting assets before the asset you don't want deleted changes the index of the target asset, it's a lot simpler to use the decrementing loop